engines/sci/graphics/frameout.cpp

```diff
@@ -55,31 +57,37 @@
 
 namespace Sci {
 
-GfxFrameout::GfxFrameout(SegManager *segMan, ResourceManager *resMan, GfxCoordAdjuster *coordAdjuster, GfxScreen *screen, GfxPalette32 *palette, GfxTransitions32 *transitions) :
+GfxFrameout::GfxFrameout(SegManager *segMan, ResourceManager *resMan, GfxCoordAdjuster *coordAdjuster, GfxPalette32 *palette, GfxTransitions32 *transitions, GfxCursor32 *cursor) :
 	_isHiRes(ConfMan.getBool("enable_high_resolution_graphics")),
 	_palette(palette),
+	_cursor(cursor),
 	_resMan(resMan),
-	_screen(screen),
 	_segMan(segMan),
 	_transitions(transitions),
 	_benchmarkingFinished(false),
 	_throttleFrameOut(true),
 	_throttleState(0),
-	// TODO: Stop using _gfxScreen
-	_currentBuffer(screen->getDisplayWidth(), screen->getDisplayHeight(), nullptr),
 	_remapOccurred(false),
 	_frameNowVisible(false),
-	_screenRect(screen->getDisplayWidth(), screen->getDisplayHeight()),
 	_overdrawThreshold(0),
 	_palMorphIsOn(false) {
 
-	_currentBuffer.setPixels(calloc(1, screen->getDisplayWidth() * screen->getDisplayHeight()));
-
-	// QFG4 is the only SCI32 game that doesn't have a high-resolution toggle
+	// QFG4 is the only SCI32 game that doesn't have a high-resolution version
 	if (g_sci->getGameId() == GID_QFG4) {
 		_isHiRes = false;
 	}
 
+	if (g_sci->getGameId() == GID_PHANTASMAGORIA) {
+		_currentBuffer = Buffer(630, 450, nullptr);
+	} else if (_isHiRes) {
+		_currentBuffer = Buffer(640, 480, nullptr);
+	} else {
+		_currentBuffer = Buffer(320, 200, nullptr);
+	}
```

Comment:
> GK1 Mac actually does not have the enable_high_resolution_graphics flag, which also makes sense when it's hires only. […]

I feel like there is a misunderstanding here.

When games are not given the `enable_high_resolution_graphics` (a.k.a. `GAMEOPTION_HIGH_RESOLUTION_GRAPHICS`) GUIO, that configuration option defaults to `true`. GK1 Mac is not given the GUIO option, so it always runs with high-resolution graphics. There’s no need to add extra GK1 Mac specific code, because it will get hires mode already, because of the way the system is designed. I hope this explanation clears things up.

> So GK1 DOS floppy defaults to low-res

It doesn’t in ScummVM at the moment; the GUIOs for all SCI32 games aren’t correct right now, which is something that needs to be addressed separately at a later time (like around the time we’re almost ready to start playtesting). I put it on my TODO list.

> Anyway are you sure that the hashes were properly taken? I mean that's the same logic. Maybe those are also bad, who knows. No one can currently check GK1 Mac. Maybe they are incomplete? How do you know that they weren't based on guesswork and are perfectly fine, when you can not verify them?

I have a somewhat higher confidence about the hashes because they’re automatically generated by code that is known to generate correct hashes, whereas I’ve been working on the SCI32 code for a while now and have a pretty low confidence that any of the old SCI32-specific code is actually correct.